scripts/workspace_status.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python3
+
+"""Bazel `--workspace_status_command` script.
+
+This script is designed to be used in Bazel`s `--workspace_status_command` and
+generate any desirable status artifacts.
+"""
```

Comment:
Note, to your point about making python a build dependency... I'll just note this could be shell, even letting bzl do the ".dirty" replacement, although I don't know if we're better off consistently using Python in place of shell (I mean, maybe this Python already requires something like WSL on Windows anyways: "On Windows this must be a native binary, typically an ".exe", ".bat", or a ".cmd" file." [here](https://bazel.build/docs/user-manual#workspace-status-command))

Reply:
My understanding is that Bash has been specifically difficult to make work here compared to Python on Windows. And so that's part of why I went with Python. I also think it's a bit easier for us to maintain. WDYT?

Reply:
Either way, I'm not sure if we'll really get out of the WSL dependency; with WSL I don't think it makes a difference, it's only if we're trying to do Windows native (which would require eliminating genrule use too).

Reply:
I'd somewhat like to someday get to Windows native support, but agree its a long way away.

I still find the Python simple enough to prefer over Bash here?